### PR TITLE
[WIP] Event-based-gamemode

### PR DESCRIPTION
**Pull request by Amelia0010**
### Intent of your Pull Request

TL;DR: CEV Eris except boiled down to a single gamemode. Like them fancy dynamic gamemodes that TG seems to be taking from VG, except this has a storyteller attached which has different point gains for different event types, meaning antag events might spawn faster with one storyteller than with the others.

Actual run-down:

I'm going to seperate all the events we currently have (and a few more that I'm gonna make specifically for this gamemode) into 4 categories, 3 based on severity, and one based on whether they make antags or not. Then a storyteller which is picked at roundstart (and which admins can change) passively generates points, and when the points in a category gets to a certain cap, they spawn an event in that category and resets it to 0

The goal is for this to make actual interplay between different antags (like what was promised with antag datums), so you can actually have nukies fighting wizards. I also intend to make it so that certain events happening on the station, like antags dieing or crewmembers dieing, also increases/decreases points, so it's at least semi-based on how well the station is doing at the time.

This is still quite a ways off, since there's a metric fucktonne of things I need to change, especially gimping different antags so it's not just gonna be a steamroll. Basically, look at my TODO list to see that it's gonna take me quite a while (especially as I'm a tad busy atm) to finish this, but hey, I like input and having an open PR makes me infinitely more likely to actually finish it than if it's just some random project on a branch on my local repo (rip spythief tho, a PR didn't save that from my burnout).

Also disregard the .dme change for now, I'll fix that (blame alexkar tho)

Nich's Grand TODO List:

1. Categorize all the random events into the 4 different categories
2. Make more antag events, basically one for every type of antag
3. Either make a 5th category for wizard events or make the summon wizard events spell just stuff em in the other 4
4. Add a random antag pick to roundstart, basically just picking one of the antag events
5. Add ways for the station to impact point generation (not directly, but things like antags dieing raising it, crewmembers dying lowering it)
6. Add the actual structure for the game_mode to run events, including game mode specific can_trigger procs for most of them, so we check whether there's an AI before we make it malf
7. Figure out what the hell I'm gonna do regarding antag rep in this gamemode, probably gonna completely disregard it
8. Overhaul uplink code so it's no longer restricted on a gamemode basis, but on an antag_datum basis
9. Figure out whether I'm gonna completely forget that hivemind/overthrow exists for this gamemode
10. Should I end the round if revs are successfull, or do I just let the station exist in a post-revolutionary world? o7
11. Maybe block conversion antags like revs and cultists from converting each other so you have both antag datums (deconversion sigil? make revflash deconvert on first use?). Could also just make it so that antags like revs and cultists stop other antags from happening while they're running, and then they all deconvert once they win?
12. Make events that are all anti-loyalty-implant (revs, gangs) mutually exclusive, so you don't get a rev event immediately after the gangs lose because everyone was converted
13. Probably gonna have to gimp a few of the gamemodes (gangs and cultists in particular) so they aren't too strong considering there'll be more than one antag group (guess I can use winrates)
14. Make a gimped version of old clock cult for this gamemode (was tempted to have new on roundstart, old as a mid-round thing, but new clock-cult kinda messes up the flow
15. Major/Antag events won't be weighted and will only happen once, while Medium/Minor events will be weighted like normal, but will slowly lose weight as they happen
16. Add DB logging of point change and events triggered so we can do proper statistics on this shit (no Altoids allowed)

#### Changelog

:cl:  
experimental: This is experimental, yes.
/:cl:

### CODE REVIEW

**Comment by Xoxeyos** [comment]
Perhaps you should add a CC alert, saying that there are no more revs, etc. It should only show up on the comms consoles.

**Comment by Xoxeyos** [comment]
Also, is this a seperate gamemode on it's on? Or will it entirely replace rounds such as only traitors, only revs, only cults, etc? It wasn't quite clear to me.

**Comment by Amelia0010** [comment]
It will be a seperate gamemode, yes.

**Comment by Laus4Deus** [comment]
I like the idea and it looks something hard but achievable

**Comment by CMOisLing** [comment]
I have no idea how this is going to work out but I am excited non the less

File: code/__DEFINES/misc.dm [hunk 1]
```diff
@@ -458,4 +458,10 @@ GLOBAL_LIST_INIT(pda_styles, list(MONO, VT, ORBITRON, SHARE))
 #define GRENADE_READY 3
 
 //Misc text define. Does 4 spaces. Used as a makeshift tabulator.
-#define FOURSPACES "&nbsp;&nbsp;&nbsp;&nbsp;"
\ No newline at end of file
+#define FOURSPACES "&nbsp;&nbsp;&nbsp;&nbsp;"
```
Comment (Moja10) [inline code/__DEFINES/misc.dm:461]: image

Graph provided by senior graph engineer Altoids

Comment (Amelia0010) [reply code/__DEFINES/misc.dm:461]: literally what

**Review by Xoxeyos — APPROVED** [review]
Ignore this, didn't realize it was incomplete

**Comment by Amelia0010** [comment]
Remind me to revive this sometime in November when I actually have time to finish this properly
